### ISSUE TITLE
I can't get this running for the life of me.

### ISSUE THREAD

**Issue by Johncunningham918** [tsightler/ring-mqtt#185]
Can someone break this guide down a bit for beginners? I have tried installing this with CLI and docker several times, and I just can't get it to work. Errors every time, and I am at my wits end. Anybody? Please?

**Comment by tsightler** [comment]
Well, this isn't exactly a tool for complete beginners so you'll likely have to put some effort to move past beginner stage before you have any success.  As docker is the preferred method to run this, and probably the easiest, I guess we should start with this method and you should share what command you are running and what errors you are getting and also share a few more details about your setup and what you are hoping to use this project to do.  Simply saying that you've tried "everything" and then not posting a single example of anything you tried or any error doesn't leave much to go on.

**Comment by Johncunningham918** [comment]
I got the hint early into this project that this path looked difficult for my experience level, but this appears to be the only way to get ring alarm devices into Home Assistant. I have tried and failed with many other methods that no longer work, which forced me to try this one in a last ditched effort. 

I am trying to run this on a standalone Pi running the raspberry pi OS in order to utilize the sensor outputs for my home assistant. Each time I have attempted was with a fresh OS install (so I wouldn't compound my failures). As of now, I am running into issues using the docker method. 

First, I update the OS to the latest versions, then install docker using "curl -sSL https://get.docker.com | sh", then "sudo usermod -aG docker pi" to set permissions. After that is complete, I enter "docker pull tsightler/ring-mqtt" and let the process finish. Followed by "docker run --rm -e "MQTTHOST=my host IP" -e "MQTTPORT=1883" -e "MQTTRINGTOPIC=ring_topic" -e "MQTTHASSTOPIC=hass_topic" -e "MQTTUSER=mqtt_broker" -e "MQTTPASSWORD=my psswd" -e "RINGTOKEN=My token generated using CLI token generation" -e "ENABLECAMERAS=false" -e "RINGLOCATIONIDS=comma-separated_location_IDs" tsightler/ring-mqtt". 

I then move onto "docker run --rm --mount type=bind,source=/etc/ring-mqtt,target=/data -e "MQTTHOST=my host IP" -e "MQTTUSER=mqtt_broker" -e "MQTTPASSWORD=my psswd" -e "RINGTOKEN=my token" tsightler/ring-mqtt", but I am met with an error. The error is "docker: Error response from daemon: invalid mount config for type "bind": bind source path does not exist: /etc/ring-mqtt." I have also tried running the docker using "docker run ring-mqtt" and get the error "Unable to find image 'ring-mqtt:latest' locally
Docker: Error response from daemon: pull access denied for ring-mqtt, repository does not exist or may require 'docker login': denied: requested access to the resource is denied."

I am not sure what I did wrong, or what exactly went wrong.

**Comment by tsightler** [comment]
Quick question before we get too deep, since you say you are using Home Assistant, are you using the standard Home Assistant install method via Home Assistant OS?  If so, you can just use the addon, no need for any manual docker stuff at all.  That's really the best path for beginners.

**Comment by Johncunningham918** [comment]
I am running the HA operating system on a Pi, and have added the ring integration from within, but it does not show any alarm devices. I don't use ring cameras, so it is just an empty and unconfigurable integration currently.

**Comment by tsightler** [comment]
Correct, the native Ring integration in Home Assistant only does cameras, so if you don't have cameras it's useless, however, this project (ring-mqtt) also has an [available addon](https://github.com/tsightler/ring-mqtt-ha-addon) which you can install easily from within Home Assistant, no command line required.

However, before installing it you'll need to make sure you have the Mosquitto addon and the Mosquitto MQTT integration working first using the instructions here:  https://github.com/home-assistant/addons/blob/master/mosquitto/DOCS.md

Basically you install the Mosquitto addon from the official addon store, then go to integrations and wait for the Mosquitto integration to be auto discovered, the click configure, make sure automatic device discovery is selected, and accept the defaults for the integration settings.

Once you have this done we can move on to installing the addon if you can't figure that part out, but it's way easier that trying to hack around on the command line.

**Comment by Johncunningham918** [comment]
I have successfully installed and configured mosquitto MQTT

**Comment by tsightler** [comment]
OK, so now you just need to add the ring-mqtt addon.  Go back to the Supervisor, Add-on Store tab, and click the three dots in the upper right hand corner and choose repositories.  Copy and past the URL to the addon and click "Add", this is the URL:

https://github.com/tsightler/ring-mqtt-ha-addon

Once this is done scroll to the bottom and you should see the Ring Devices via MQTT addon.  Install the addon and start it (for alarms you shouldn't need any configuratoin).  Give it a 10-30 seconds and then use the web UI to authenticate to ring and automatically create the authentication token.  Wait a few minutes and your Ring devices should start appearing.

**Comment by Johncunningham918** [comment]
Wow. I can't believe how easy that was. How did I miss this.... I only got 3 hours of sleep last night because I told myself I would rerun the codes on the github again and again until I figured it out. Man do I feel stupid. Thank you so very very much for your help. I can't tell you how much I appreciate it. Do you have a tip service/buymeacoffee setup, or some other way to thank you for your time/help?

**Comment by tsightler** [comment]
The project was not originally built for Home Assistant specifically, but I tried to make the addon as easy as possible to get it running in HA since these days most of the users are using Home Assistant and I've found a lot of HA users aren't super deep with Linux skills.  Sounds like you got it working and I'm glad to hear that!

Regarding tip/buymeacoffee, I don't have any such setup and I would likely just donate the money anyway, so I always say just cut out the middleman and donate anything you would tip to something you care about, if you have a desire to do something, but of course, no obligation to do anything.

**Comment by Johncunningham918** [comment]
You name the charity and I'll make it happen! Thank you again for your help!

**Comment by tsightler** [comment]
The majority of my giving is local and private because I feel like the smaller the organization the less wasted on bureaucracy, but, if you wanted to donate to something national I'd have to say that the National MS Society, while not a perfect organization, is special to me due to the work that they do and the fact that one of my children has MS so their resources and efforts are important to me personally.

However, there are plenty of great charities out there and I always leave it to the user what they would like to do if anything at all.  People have sent me notes from all types of charities from all over the world that I didn't even know about, orphanages, food pantries during the worst of COVID, etc.  It makes me happy to know that this silly project has somehow had even the slightest positive impact in the world outside of letting people automatically arm their alarm when they are gone or turn the lights on when they get home!  :)

**Comment by Johncunningham918** [comment]
I'm sorry to hear about your son. Thank you for the work that you do, and best of luck on your future endeavors! I made the donation in your honor (TSightler)!
